prov/efa/src/efa_prov.c

```diff
@@ -232,7 +235,10 @@ EFA_INI
 	if (err)
 		goto err_free;
 
+	ofi_mutex_init(&g_efa_domain_list_lock);
+	ofi_mutex_lock(&g_efa_domain_list_lock);
```

Comment:
This part is actually protected by the ini lock https://github.com/ofiwg/libfabric/blob/5dd902300d2c5cdf27f436ffe430d4217c093f5c/src/fabric.c#L926

It seems weird to me to initialize the lock and then immediately grab it. Because then who is protecting against a race condition in the lock initialization?